app/containers/Admin/ApplicationReview/ReviewSidebar.tsx

```diff
@@ -0,0 +1,343 @@
+import {graphql, createFragmentContainer, RelayProp} from 'react-relay';
+import {ReviewSidebar_applicationReviewStep} from '__generated__/ReviewSidebar_applicationReviewStep.graphql';
+import ReviewComment from 'components/Admin/ReviewComment';
+import updateReviewCommentMutation from 'mutations/application_review_step/updateReviewCommentMutation';
+import deleteReviewCommentMutation from 'mutations/application_review_step/deleteReviewCommentMutation';
+import updateApplicationReviewStepMutation from 'mutations/application_review_step/updateApplicationReviewStepMutation';
+import {nowMoment} from 'functions/formatDates';
+import {capitalize} from 'lib/text-transforms';
+
+interface Props {
+  onClose: () => void;
+  applicationReviewStep: ReviewSidebar_applicationReviewStep;
+  relay: RelayProp;
+}
+
+export const ReviewSidebar: React.FunctionComponent<Props> = ({
+  onClose,
+  applicationReviewStep,
+  relay
+}) => {
+  const [showingResolved, setShowingResolved] = useState(false);
+  const toggleResolved = () => setShowingResolved((current) => !current);
+  const reviewStepName = capitalize(
+    applicationReviewStep.reviewStepByReviewStepId.stepName
+  );
+
+  const generalComments = applicationReviewStep.generalComments.edges;
+  const internalComments = applicationReviewStep.internalComments.edges;
+  const noGeneralCommentsToShow =
+    generalComments.length === 0 ||
+    (!showingResolved && generalComments.every((c) => c.node.resolved));
+  const noInternalCommentsToShow =
+    internalComments.length === 0 ||
+    (!showingResolved && internalComments.every((c) => c.node.resolved));
+
+  const markReviewStepComplete = async (isComplete) => {
+    const {environment} = relay;
+    const variables = {
+      input: {
+        id: applicationReviewStep.id,
+        applicationReviewStepPatch: {
+          isComplete
+        }
+      }
+    };
+    await updateApplicationReviewStepMutation(environment, variables);
+  };
+  const markCompletedButton = (
+    <Button
+      id="markCompleted"
+      variant="outline-primary"
+      type="button"
+      onClick={() => markReviewStepComplete(true)}
+      style={{
+        padding: '0.75rem .9rem',
+        display: 'block',
+        margin: 'auto'
+      }}
+    >
+      <FontAwesomeIcon icon={faCheck} style={{marginRight: '0.5rem'}} />
+      Mark this review step completed
+    </Button>
+  );
+  const resolveComment = async (id, resolve) => {
+    const {environment} = relay;
+    const variables = {
+      input: {
+        id,
+        reviewCommentPatch: {
+          resolved: resolve
+        }
+      }
+    };
+    await updateReviewCommentMutation(environment, variables);
+  };
+  const deleteComment = async (id, commentType) => {
+    const {environment} = relay;
+    const variables = {
+      input: {
+        id,
+        reviewCommentPatch: {
+          deletedAt: nowMoment().format('YYYY-MM-DDTHH:mm:ss')
+        }
```

Comment:
I don't think that this patch is needed. There is an issue with `deleteReviewCommentMutation` where the input type is wrong. The only input should be the `id`

Reply:
> There is an issue with deleteReviewCommentMutation where the input type is wrong.

Could you elaborate?

Reply:
I was referring to my comment in `deleteReviewCommentMutation`, but it turns out I was wrong, so nevermind :)